### ISSUE TITLE
Pass in LUTs to lookup ufunc

### ISSUE THREAD

**Issue by mhostetter** [mhostetter/galois#79]
Only compile lookup ufuncs once and pass in lookup tables as arguments.

**Comment by mhostetter** [comment]
If this is done, then `cache=True` should be used and the ufuncs should be compiled for each dtype.

```python
        one_arg_signatures = []
        two_arg_signatures = []
        for dtype in cls.dtypes:
            d = np.dtype(dtype).name
            one_arg_signatures.append(f"{d}({d})")
            two_arg_signatures.append(f"{d}({d}, {d})")

        # Create numba JIT-compiled ufuncs using the *current* EXP, LOG, and MUL_INV lookup tables
        cls._ufuncs["add"] = numba.vectorize(two_arg_signatures, **kwargs)(_add_lookup)
        cls._ufuncs["subtract"] = numba.vectorize(two_arg_signatures, **kwargs)(_subtract_lookup)
        cls._ufuncs["multiply"] = numba.vectorize(two_arg_signatures, **kwargs)(_multiply_lookup)
        cls._ufuncs["divide"] = numba.vectorize(two_arg_signatures, **kwargs)(_divide_lookup)
        cls._ufuncs["negative"] = numba.vectorize(one_arg_signatures, **kwargs)(_additive_inverse_lookup)
        cls._ufuncs["reciprocal"] = numba.vectorize(one_arg_signatures, **kwargs)(_multiplicative_inverse_lookup)
        cls._ufuncs["scalar_multiply"] = numba.vectorize(two_arg_signatures, **kwargs)(_scalar_multiply_lookup)
        cls._ufuncs["power"] = numba.vectorize(two_arg_signatures, **kwargs)(_power_lookup)
        cls._ufuncs["log"] = numba.vectorize(one_arg_signatures, **kwargs)(_log_lookup)
        cls._ufuncs["poly_eval"] = numba.guvectorize([(numba.int64[:], numba.int64[:], numba.int64[:])], "(n),(m)->(m)", **kwargs)(_poly_eval_lookup)
```

**Comment by mhostetter** [comment]
Passing in LUTs (matrix arguments) is only supported with `guvectorize` which is slower than `vectorize`. Passing in scalars like `characteristic, degree, irreducible_poly` for "calculate" ufuncs is possible with `vectorize`, however those ufuncs would now take 5 arguments and only binary (2 argument) ufuncs support ufunc methods like `outer`. So doing so would forfeit all ufunc methods.

I've tried doing this multiple ways and haven't found a reasonable solution. Closing for now.
